tests/test_profiling_result_printer.py

```diff
@@ -0,0 +1,163 @@
+from typing import Any
+
+import pytest
+import numpy
+
+import modmesh
+from modmesh.profiling import (
+    ProfilingColumnData,
+    ProfilingTableBuilder,
+    ProfilingResultPrinter,
+)
+
+
+class TestProfilingResultPrinter:
+    def run_profile_function(self):
+        _ = modmesh.CallProfilerProbe("numpy_arange_100")
+        numpy.arange(0, 100, dtype="uint32")
+
+    @pytest.fixture
+    def profiling_result_fixture(self) -> dict[str, Any]:
+        modmesh.call_profiler.reset()
+
+        self.run_profile_function()
+
+        result: dict[str, Any] = modmesh.call_profiler.result()["children"]
+        return result
+
+    def test_printer_getitem_with_valid_key_should_return_correct_data(
+        self, profiling_result_fixture: dict[str, Any]
+    ) -> None:
+        printer: ProfilingResultPrinter = ProfilingResultPrinter(
+            profiling_result_fixture
+        )
+
+        assert printer["numpy_arange_100"].name == "numpy_arange_100"
+
+    def test_printer_getitem_with_absent_key_should_raise_value_error(
+        self, profiling_result_fixture: dict[str, Any]
+    ) -> None:
```

Comment:
I don't like the specification of return type in Python.  It usually confuses more than clarifies.  You can still do it but keep it in mind.